web/modules/custom/nidirect_webforms/src/Plugin/WebformHandler/PrisonVisitBookingHandler.php

```diff
@@ -265,12 +268,37 @@ public function alterForm(array &$form, FormStateInterface $form_state, WebformS
     $form['#attached']['drupalSettings']['prisonVisitBooking'] = $this->configuration;
     $form['#attached']['drupalSettings']['prisonVisitBooking']['booking_ref'] = $this->bookingReference;
 
+    // Detect if booking reference (visit_order_number) is changed.
+    $last_ref = $form_state->get('last_visitor_order_number');
+    $curr_ref = $form_state->getValue('visitor_order_number');
+
+    // First-time set.
+    if ($last_ref === NULL && !empty($curr_ref)) {
```

Comment:
Safer to check for non-values with `empty()` instead of NULL, which is a far more specific type and (non)-value check